### ISSUE TITLE
训练的时候使用 cpu

### ISSUE THREAD

**Issue by Hua-Wu-Que-Code** [RUCAIBox/R1-Searcher#35]
我只有一个 gpu 设备，在运行 ray 服务时，我修改代码为：
`# # source /opt/aps/workdir/input/jiechen/.venv/bin/activate
export CUDA_VISIBLE_DEVICES=0

# conda activate openrlhf

HEAD_NODE_IP=127.0.0.1

ray start --head --node-ip-address ${HEAD_NODE_IP} --num-gpus 1 --port 8266 --dashboard-port 8267

`
运行` ray status `确实给 ray 服务分配了gpu
`======== Autoscaler status: 2025-04-26 00:50:55.329580 ========
Node status
---------------------------------------------------------------
Active:
 1 node_bee267da493348d498f97a1263f4d5dc15d90b7b88806e1c3cc21858
Pending:
 (no pending nodes)
Recent failures:
 (no failures)

Resources
---------------------------------------------------------------
Usage:
 0.0/16.0 CPU
 0.0/1.0 GPU
 0B/71.52GiB memory
 0B/34.64GiB object_store_memory

Demands:`
然后修改`qwen_reinforce_plus_train.sh`文件为
`# 设置环境变量
export NUMEXPR_MAX_THREADS=128  # 设置 NumExpr 的最大线程数，用于优化计算性能
export RAY_DEDUP_LOGS=0         # 禁用 Ray 的重复日志输出

# WandB 配置
#wandb_token=xxx                 # WandB 的 API Token，用于记录训练日志
#export WANDB_API_KEY=xxx        # 设置 WandB 的 API Key 环境变量

# 数据路径
DATA_PATH=/root/autodl-tmp/R1-Searcher/wiki_corpus_index_bulid/samples/train_kilt_full_page_sample_100.tsv
# 指定训练数据的路径，通常是一个 TSV 文件，包含训练所需的输入数据

# 模型路径
TOKENIZER_PATH=/root/autodl-tmp/R1-Searcher/model/qwen2.5
# 指定预训练模型的路径，用于初始化训练

# GPU 配置
export CUDA_VISIBLE_DEVICES=0   # 指定使用的 GPU（这里是单 GPU，编号为 0）

# 训练参数
N_SAMPLES=16                    # 每个提示生成的样本数量
EPISODE=10000                   # 训练的总 Episode 数
WARMUP=0.0                      # 学习率预热比例
TBS=32                          # 训练的 batch size
RBS=8                           # rollout 的 batch size
KL=0.0                          # KL 散度系数，用于正则化
LR=2e-6                         # Actor 模型的学习率
MAX_LENGTH=29000                # 生成的最大序列长度
PORT=1278                       # Reward Server 的端口号
TEMP=1.0                        # 生成时的温度参数，用于控制生成的多样性

# 模型保存名称
SAVE_MODEL_NAME=qwen2.5         # 保存模型的名称，用于组织训练结果

# 其他配置
GROUP_METHOD=normal             # 分组方法（具体含义取决于代码实现）
LOG_BASE=log                    # 日志文件的基础路径

# 创建保存结果的目录
mkdir -p /root/autodl-tmp/R1-Searcher/results/$SAVE_MODEL_NAME
mkdir -p /root/autodl-tmp/R1-Searcher/results/ckpts
mkdir -p /root/autodl-tmp/R1-Searcher/results/$SAVE_MODEL_NAME/server
mkdir -p $LOG_BASE/server/

# 提交 Ray 任务
ray job submit --address="http://127.0.0.1:8267" \
   -- python3 -m openrlhf.cli.train_ppo_ray \
   --ref_num_nodes 1 \
   --ref_num_gpus_per_node 1 \
   --actor_num_nodes 1 \
   --actor_num_gpus_per_node 1 \
   --vllm_num_engines 1 \
   --vllm_tensor_parallel_size 1 \
   --pretrain ${TOKENIZER_PATH} \
   --remote_rm_url http://localhost:${PORT}/get_reward \
   --save_path /root/autodl-tmp/R1-Searcher/results/ckpts/$SAVE_MODEL_NAME \
   --ckpt_path /root/autodl-tmp/R1-Searcher/results/ckpts/$SAVE_MODEL_NAME \
   --micro_train_batch_size 1 \
   --train_batch_size ${TBS} \
   --micro_rollout_batch_size 1 \
   --rollout_batch_size ${RBS} \
   --advantage_estimator reinforce_baseline \
   --max_samples 100000 \
   --max_epochs 1 \
   --num_episodes ${EPISODE} \
   --lr_warmup_ratio ${WARMUP} \
   --n_samples_per_prompt $N_SAMPLES \
   --prompt_max_len 1024 \
   --generate_max_len $MAX_LENGTH \
   --zero_stage 2 \
   --bf16 \
   --load_checkpoint \
   --actor_learning_rate $LR \
   --critic_learning_rate 9e-6 \
   --init_kl_coef $KL \
   --prompt_data $DATA_PATH \
   --input_key question \
   --flash_attn \
   --gradient_checkpointing \
   --save_steps 20 \
   --vllm_sync_backend nccl \
   --max_ckpt_num 3 \
   --temperature $TEMP \
   --overlap_comm \
   --packing_samples`
再执行命令`bash scripts/qwen_reinforce_plus_train.sh | tee results/logs/qwen_reinforce_plus_t`
发现训练时，只在 cpu 上运行而非 gpu

![Image](https://github.com/user-attachments/assets/4dbfe3c7-b7c8-4ccc-b8ae-da55a19b3bcd)

![Image](https://github.com/user-attachments/assets/8d10466e-93c7-4dc3-8a3f-05f6ff01c84e)
不知我哪里有错误呢？

**Comment by SsmallSong** [comment]
1张gpu无法训练，因为vllm也需要gpu。你的启动命令里实际上是需要两张gpu

**Comment by Hua-Wu-Que-Code** [comment]
不好意思，再次打扰您一下，我看到了`wandb_token`，但是我查找了很多资料都没有看到这个是如何获取的，更多的都是`WANDB_API_KEY`，您可以教一下吗？
